File: cyber/demo_py/demo01_listener.py
from cyber.python.cyber_py3 import cyber
from cyber.demo_base_proto.student_pb2 import Student

def callback(student):
    print(student.age)


if __name__ == "__main__":
    cyber.init()
    listener_node  = cyber.Node("lis")
    listener = listener_node.create_reader("chatter", Student, callback)
    listener_node.spin()
    cyber.shutdown()

File: cyber/demo_cc/demo01_listener.cc
#include "cyber/cyber.h"
#include "cyber/demo_base_proto/student.pb.h"

using apollo::cyber::demo_base_proto::Student;


void CallBack(const std::shared_ptr<apollo::cyber::demo_base_proto::Student> &msg){
    AINFO << "receive " << msg->age();
}

int main(int argc, char const *argv[])
{
    apollo::cyber::Init(argv[0]);

    auto listener_node = apollo::cyber::CreateNode("wuxing_go");
    auto listener = listener_node->CreateReader<Student>("chatter", CallBack);


    apollo::cyber::WaitForShutdown();
    return 0;
}